src/startUp.ts

```diff
@@ -26,12 +26,19 @@ const checkArgs = (args: IArgConfig) => {
         console.error("Options 'rate-limit-global-windowms' and 'rate-limit-global-max' must be provided together.");
         process.exit(1);
     }
+
+    if (commandLineConfig.loglevel && !Object.values(LogLevel).includes(commandLineConfig.loglevel)) {
+        console.error("Options 'loglevel' can only be 'error', 'info' or 'debug'.");
+        process.exit(1);
+    }
 };
 
 async function startUp() {
     checkArgs(commandLineConfig);
     const config = Object.assign(jsonConfig, commandLineConfig);
 
+    setLogLevel(config.loglevel as LogLevel);
```

Comment:
After using tryParse above, you will no longer need this cast.

Reply:
Well, config is IArgConfig, so config.loglevel is still just a string.
Instead, I changed setLogLevel to accept a string and throw ApplicationError if a wrong level is passed.